src/record/record-handler.js

```diff
@@ -106,6 +120,47 @@ RecordHandler.prototype.handle = function( socketWrapper, message ) {
 };
 
 /**
+ * Tries to retrieve the record from the cache or storage. If not found in either
+ * returns false, otherwise returns true
+ * @param   {SocketWrapper} socketWrapper the socket that send the request
+ * @param   {Object} message parsed and validated message
+ *
+ * @private
+ * @returns {void}
+ */
+RecordHandler.prototype._hasRecord = function( socketWrapper, message ) {
+	var recordName = message.data[ 0 ],
+		onComplete = function( record ) {
+			var hasRecord = record ? 'T' : 'F';
+			socketWrapper.sendMessage( C.TOPIC.RECORD, C.ACTIONS.HAS, [ recordName, hasRecord ] );
+		};
+
+	new RecordRequest( recordName, this._options, socketWrapper, onComplete.bind( this ) );
```

Comment:
Do we rely on the request timing out if an error occurred? Maybe rather pass an error callback.